src/lib.rs

```diff
@@ -150,14 +150,20 @@ impl<'a> From<&'a str> for Host {
 #[derive(Debug, Clone, PartialEq, Eq, Hash, Deserialize)]
 #[serde(from = "String")]
 pub enum Target {
-    BuiltIn { triple: String },
-    Custom { triple: String },
+    BuiltIn {
+        triple: String,
+        libc: Option<String>,
```

Comment:
> this seems wierd, think there's a better way to encapsulate it

How? We split the triple for built-in images, which are the ones where we can add targets for, into the actual triple and the libc version, if present. The serialized triple is the same for the docker command, so it correctly has the libc version when passed to `zigbuild`, but for rustup and adding targets, etc., it uses the regular triple.

Reply:
The reason I think it's wierd is because it's not real syntax compatible with cargo.

I don't think the version should be a property of the target, as it's not part of the target string outside zig.

If we use this way, there's no good way to specify glibc without always specifying the version in `--target`.

Instead, I think the version specifier should be a property in the config, possibly

```toml
[target.$target.zig]
version = "2.17"
```

and `Target` should not have the version encoded.

the big thing I'm seeing is that 

1. --target $target.$glibc is not a real target according to rustc/cargo
2. you'll have to always specify the version number when using `*.zig` to be correct
3. the glibc version becomes promoted, i.e always forwarded, even if it's only used in one case.